extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/devmode/DevModeStreamsCollector.java

```diff
@@ -20,7 +22,8 @@
     }
 
     public void shutdown() {
-        streamObservers.forEach(this::complete);
+        List<StreamObserver<?>> observers = new ArrayList<>(streamObservers);
```

Comment:
I don't believe it will be enough. You need to synchronize the access to streamObservers and create the copy in the synchronized block.

Reply:
the chance of hitting is is rather thin, but true, it can happen that someone connects to the stream just at the time of reload it won't work proerly. I'll change this

Reply:
done